library/init/data/nat/lemmas.lean

```diff
@@ -730,5 +770,148 @@ lemma pred_inj : ∀ {a b : nat}, a > 0 → b > 0 → nat.pred a = nat.pred b 
 
 /- TODO(Leo): sub + inequalities -/
 
+/- div -/
+
+theorem zero_div : ∀ m : ℕ, 0 / m = 0
+  | zero := by refl
+  | (succ n) := by refl
+
+theorem div_le_div
+: ∀ {y x}, x ≤ y → ∀ z : ℕ, x / z ≤ y / z :=
```

Comment:
I think I see what you mean. You're suggesting I strengthen the theorem to an equivalence (with z \ne 0 as a side condition). I'll try adding 

theorem div_le_div_of_le
: ∀ z : ℕ, z ≠ 0 → x / z ≤ y / z → x ≤ y

and

theorem div_le_div_iff_le
: ∀ z : ℕ, z ≠ 0 → ( x / z ≤ y / z ↔ x ≤ y )

and get back to you.